XamarinCommunityToolkit/Helpers/WeakEventManager.shared.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+using static System.String;
+
+namespace Xamarin.CommunityToolkit.Helpers
+{
+	// Copy from https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Core/WeakEventManager.cs
+	// TODO: Remove when PR complete: https://github.com/xamarin/Xamarin.Forms/pull/12063
+	class WeakEventManager
+	{
+		readonly Dictionary<string, List<Subscription>> eventHandlers = new Dictionary<string, List<Subscription>>();
+
+		public void AddEventHandler<TEventArgs>(EventHandler<TEventArgs> handler,
+			[CallerMemberName] string eventName = null)
+			where TEventArgs : EventArgs
+		{
+			if (IsNullOrEmpty(eventName))
+				throw new ArgumentNullException(nameof(eventName));
+
+			if (handler == null)
+				throw new ArgumentNullException(nameof(handler));
+
+			AddEventHandler(eventName, handler.Target, handler.GetMethodInfo());
+		}
+
+		public void AddEventHandler(EventHandler handler, [CallerMemberName] string eventName = null)
+		{
+			if (IsNullOrEmpty(eventName))
+				throw new ArgumentNullException(nameof(eventName));
+
+			if (handler == null)
+				throw new ArgumentNullException(nameof(handler));
+
+			AddEventHandler(eventName, handler.Target, handler.GetMethodInfo());
+		}
+
+		public void HandleEvent(object sender, object args, string eventName)
+		{
+			var toRaise = new List<(object subscriber, MethodInfo handler)>();
+			var toRemove = new List<Subscription>();
+
+			if (eventHandlers.TryGetValue(eventName, out var target))
+			{
+				for (var i = 0; i < target.Count; i++)
```

Comment:
Instead of using the property directly can you use a local variable?
```suggestion
        var targetCount = target.Count;
				for (var i = 0; i < targetCount; i++)
```
On my tests, this change will be 12% faster(running on Mono), so it's an easy win for us.
You can try it [here](https://github.com/pictos/XamarinCommunityToolkit/tree/bechmark)
Please, do this change for others `for` calls.

Reply:
@brminnick when you port your implementation, please take this in consideration <3